target/scripts/build/packages.sh

```diff
@@ -0,0 +1,101 @@
+#! /bin/bash
+
+set -eE -u -o pipefail
+
+# shellcheck source=../helpers/log.sh
+source /usr/local/bin/helpers/log.sh
+
+export DEBIAN_FRONTEND=noninteractive
+QUIET='-qq' ; _log_level_is_trace && QUIET='-y'
+
+function _pre_installation_steps() {
+  _log 'info' 'Starting package installation'
+  _log 'debug' 'Running pre-installation steps'
+
+  _log 'trace' 'Updating package signatures'
+  apt-get "${QUIET}" update
+
+  _log 'trace' 'Installing packages that are needed early'
+  apt-get "${QUIET}" install --no-install-recommends \
+    apt-transport-https apt-utils ca-certificates \
+    curl gnupg2 ssl-cert 2>/dev/null
+
+  _log 'trace' 'Upgrading packages'
+  apt-get "${QUIET}" dist-upgrade
+
+  _log 'trace' 'Updating CA certificates'
+  update-ca-certificates
+}
+
+function _install_postfix() {
+  _log 'debug' 'Installing Postfix'
+
+  _log 'warn' "Applying workaround for Postfix bug (see https://github.com/docker-mailserver/docker-mailserver/issues/2023#issuecomment-855326403)"
+
+  mv /bin/hostname /bin/hostname.bak
+  echo "echo 'docker-mailserver.invalid'" >/bin/hostname
+  chmod +x /bin/hostname
```

Comment:
Just as a note, I think `docker build` offers a `--hostname` arg now which might allow to remove the workaround. Don't worry about that for this PR though.